xbmc/pvr/channels/PVRChannelsPath.cpp

```diff
@@ -0,0 +1,161 @@
+/*
+ *  Copyright (C) 2012-2018 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "PVRChannelsPath.h"
+
+#include "URL.h"
+#include "utils/StringUtils.h"
+#include "utils/URIUtils.h"
+#include "utils/log.h"
+
+#include <string>
+#include <vector>
+
+using namespace PVR;
+
+const std::string CPVRChannelsPath::PATH_TV_CHANNELS = "pvr://channels/tv/";
+const std::string CPVRChannelsPath::PATH_RADIO_CHANNELS = "pvr://channels/radio/";
+
+
+CPVRChannelsPath::CPVRChannelsPath(const std::string& strPath)
+{
+  std::string strVarPath = TrimSlashes(strPath);
+  const std::vector<std::string> segments = URIUtils::SplitPath(strVarPath);
+
+  for (const std::string& segment : segments)
+  {
+    switch (m_kind)
+    {
+      case Kind::INVALID:
```

Comment:
> Do not indent `case` and `default`.

https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#332-switch-case

Reply:
I'd like to discuss this rule. Every "do not indent next line after an opening  curly brace" looks ugly and unnatural to me.

Reply:
Yes, I agree. And I think we never did this previously. Furthermore, since it feels unnatural, it's hard to remember. I also don't understand how it ended up in https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md

Reply:
I'm fine with discussing this, but currently it violates the coding guidelines.

Reply:
It's strange that this ended up in the coding guidelines while we've been doing it differently all of the place for many years. I'm pretty sure the current codebase mainly uses the style preferred by @ksooo and myself. So I think we need to agree on what to do before we "taint" it any further.